src/wfs/src/main/java/org/vfny/geoserver/wfs/servlets/TestWfsPost.java

```diff
@@ -303,4 +316,24 @@ protected void processRequest(HttpServletRequest request, HttpServletResponse re
             }
         }
     }
+
+    private void validateURL(HttpServletRequest request, String url) {
+        String proxyBase = GeoServerExtensions.getProperty(PROXY_BASE_URL);
+        if(proxyBase != null) {
+            if(!url.startsWith(proxyBase)) {
+                throw new IllegalArgumentException("Invalid url requested, the demo requests should be hitting: " + proxyBase);
+            }
+        }
+        
+        // use the requested url then, and remove the TestWfsPort
+        String requestedUrl = request.getRequestURL().toString();
+        // this should not happen, but let's not make it an open proxy if it does
+        if(!requestedUrl.endsWith(TEST_WFS_POST_PATH)) {
+            throw new IllegalStateException("Unepected, the TestWfsPost was accessed by a path not ending with TestWfsPost: " + requestedUrl);
+        }
```

Comment:
While I agree that this servlet has not moved in 10+ years - I am not sure I am getting why this check is required/useful? I was more focused on the urlString (since that is what we are opening up a connection to).